backend/fms_core/tests/test_resources.py

```diff
@@ -595,3 +599,57 @@ def test_invalid_container_rename(self):
                 self._test_invalid_rename_template(rf, err)
 
             transaction.savepoint_rollback(s)
+
+    def create_projects(self):
+        self.project1, _ = Project.objects.get_or_create(name="TestProject1", status="Ongoing")
+        self.project2, _ = Project.objects.get_or_create(name="TestProject2", status="Ongoing")
+
+    def link_projects_to_samples(self):
+        self.load_samples()
+        self.create_projects()
+
+        with reversion.create_revision(), open(PROJECT_LINK_SAMPLES_CSV) as sf:
+            s = Dataset().load(sf.read())
+            self.pr.import_data(s, raise_errors=True)
+
+            reversion.set_comment("Linked projects with samples")
+
+    def test_link_projects_to_samples_import(self):
+        self.link_projects_to_samples()
+        sample1 = Sample.objects.get(name="sample1", container__barcode="tube001")
+        sample2 = Sample.objects.get(name="sample2", container__barcode="tube002")
+
+        # Test basic import success
+        self.assertEqual(len(SampleByProject.objects.all()), 2)
+
+        # Test parent record auto-generation
+        self.assertTrue(SampleByProject.objects.filter(sample=sample1, project=self.project1).exists())
+        self.assertTrue(SampleByProject.objects.filter(sample=sample2, project=self.project2).exists())
+
+    def unlink_projects_to_samples(self):
+        self.link_projects_to_samples()
+        with reversion.create_revision(), open(PROJECT_UNLINK_SAMPLES_CSV) as sf:
+            s = Dataset().load(sf.read())
+            self.pr.import_data(s, raise_errors=True)
+
+            reversion.set_comment("Unlinked projects with samples")
+
+    def test_unlink_projects_to_samples_import(self):
+        self.unlink_projects_to_samples()
+        sample1 = Sample.objects.get(name="sample1", container__barcode="tube001")
+        sample2 = Sample.objects.get(name="sample2", container__barcode="tube002")
+
+        # Test basic import success
+        self.assertEqual(len(SampleByProject.objects.all()), 1)
+
+        # Test parent record auto-generation
+        self.assertFalse(SampleByProject.objects.filter(sample=sample1, project=self.project1).exists())
+        self.assertTrue(SampleByProject.objects.filter(sample=sample2, project=self.project2).exists())
+
+    def test_invalid_link_project_sample(self):
+        self.load_samples()
+        self.create_projects()
+
+        with self.assertRaises(ValidationError), open(TEST_DATA_ROOT / "Project_link_samples_invalid_project_and_sample.csv") as sf:
+            s = Dataset().load(sf.read())
+            self.pr.import_data(s, raise_errors=True)
```

Comment:
Does this test work for you? On my end it fails with error: django.core.exceptions.ValidationError: {'sample': ['No sample with name [InvalidSample] at those container coordinates.']}

Reply:
pull ;)